### PR TITLE
build: bump btcsuite and neutrino dependencies to latest tagged versions

### PR DESCRIPTION
**Pull request by wpaulino**
The btcwallet update includes some edge-case wallet bug fixes and an
optimization on ZMQ connections for bitcoind backends.

The btcd update allows for compatibility with bitcoind v0.19.0 backends.

Fixes https://github.com/lightningnetwork/lnd/issues/2961.
Fixes https://github.com/lightningnetwork/lnd/issues/3412
Fixes https://github.com/lightningnetwork/lnd/issues/3625.
Fixes https://github.com/lightningnetwork/lnd/issues/3640.

### CODE REVIEW

**Comment by cfromknecht** [comment]
also change build script to pull in bitcoind 0.19?

**Comment by wpaulino** [comment]
> also change build script to pull in bitcoind 0.19?

Definitely, I plan to do so once Travis is restored. I'll push another commit to pull `0.19.0` so we can have both builds (`0.18.1` and `0.19.0`) show under the same PR.

**Comment by Roasbeef** [comment]
Can be rebased now that the dependent PRs are merged. Perhaps we should also update the neutrino dep as well?

**Comment by wpaulino** [comment]
Rebased! Tests still fail with bumping Travis to use `v0.19.0` because it hasn't been officially released yet. We'll want to wait until that's done and re-run the test suite to ensure nothing has changed (test suite passes as is locally).

**Comment by wpaulino** [comment]
@Roasbeef @halseth as discussed, we'll be postponing the Travis upgrade to `bitcoind 0.19.0` until the binaries are out, so we can move forward with this PR.

File: go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ require (
 	github.com/NebulousLabs/fastrand v0.0.0-20181203155948-6fb6489aac4e // indirect
 	github.com/NebulousLabs/go-upnp v0.0.0-20180202185039-29b680b06c82
 	github.com/Yawning/aez v0.0.0-20180114000226-4dad034d9db2
-	github.com/btcsuite/btcd v0.20.0-beta
+	github.com/btcsuite/btcd v0.20.0-beta.0.20191109022137-037489ef505e
```
Comment (Roasbeef) [inline go.mod]: Should update:  
  * `btcd` to `v0.20.1`
  * `neutrino` to `v0.11.0`
  * `btcwallet` to `v0.11.0`

Comment (wpaulino) [reply go.mod]: Done.

**Pull request merged**